tests/test_routing.py

```diff
@@ -282,6 +282,52 @@ def profile_page(username: str) -> H1:
     assert response.text == "<h1>Profile URL: /profile/johndoe</h1>"
 
 
+def test_air_router_url_helper_supports_query_params() -> None:
+    app = air.Air()
+    router = air.AirRouter()
+
+    @router.get("/items/{item_id}")
+    def get_item(
+        item_id: int,
+        tags: list[str] | None = air.Query(None),  # noqa: B008
+        page: int = 1,
+    ) -> air.P:
+        return air.P(f"Item {item_id} tags {tags} page {page}")
+
+    app.include_router(router)
+
+    url = get_item.url(item_id=5, query_params={"tags": ["a", "b"], "page": 2})
+    assert url == "/items/5?tags=a&tags=b&page=2"
+
+    client = TestClient(app)
+    response = client.get(url)
+    assert response.status_code == 200
+    assert response.text == "<p>Item 5 tags [&#x27;a&#x27;, &#x27;b&#x27;] page 2</p>"
```

Comment:
The test doesn't expect raw single quotes `['a', 'b']` because Air correctly escapes them to `[&#x27;a&#x27;, &#x27;b&#x27;]`. Browsers automatically decode those entities, so the UI still displays `['a', 'b']`.